packages/client/tests/functional/interactive-transactions/tests.ts

```diff
@@ -338,434 +410,471 @@ testMatrix.setupTestSuite(({ provider }, _suiteMeta, clientMeta) => {
-      await prisma.$transaction(
-        async (tx) => {
-          await tx.user.create({ data: { email: 'user@example.com' } })
-        },
-        {
-          // @ts-test-if: !['mongodb', 'sqlite', 'cockroachdb'].includes(provider)
-          isolationLevel: Prisma.TransactionIsolationLevel.RepeatableRead,
-        },
-      )
-      await expect(prisma.user.findMany()).resolves.toHaveLength(1)
-    })
+      // Skip on dataproxy because lack of logging here yet
+      testIf(!clientMeta.dataProxy)(
+        'should log queries that are executed inside an interactive transaction',
+        async () => {
+          const client = newPrismaClient({
+            log: [
+              {
+                emit: 'event',
+                level: 'query',
+              },
+            ],
+          })
 
-    testIsolationLevel('serializable', true, async () => {
-      await prisma.$transaction(
-        async (tx) => {
-          await tx.user.create({ data: { email: 'user@example.com' } })
-        },
-        {
-          // @ts-test-if: provider !== 'mongodb'
-          isolationLevel: Prisma.TransactionIsolationLevel.Serializable,
-        },
-      )
+          const didLogCommit = new Promise((resolve) => {
+            client.$on('query', () => {
+              // If any query is emmited we know that we are logging
+              resolve(true)
+            })
+          })
 
-      await expect(prisma.user.findMany()).resolves.toHaveLength(1)
-    })
+          await client.$transaction(async (tx) => {
+            const user = await tx.user.create({
+              data: {
+                email: faker.internet.email(),
+              },
+            })
+
+            await tx.user.findFirst({
+              where: {
+                id: user.id,
+              },
+            })
+          })
 
-    // TODO: there is also Snapshot level for sqlserver
-    // it needs to be explicitly enabled on DB level and test setup can't do it at the moment
-    // ref: https://docs.microsoft.com/en-us/troubleshoot/sql/analysis-services/enable-snapshot-transaction-isolation-level
-    // testIsolationLevel('snapshot', provider === 'sqlserver', async () => {
-    //   await prisma.$transaction(
-    //     async (tx) => {
-    //       await tx.user.create({ data: { email: 'user@example.com' } })
-    //     },
-    //     {
-    //       // @ts-test-if: provider === 'sqlserver'
-    //       isolationLevel: Prisma.TransactionIsolationLevel.Snapshot,
-    //     },
-    //   )
-
-    //   await expect(prisma.user.findMany()).resolves.toHaveLength(1)
-    // })
-
-    test('invalid value', async () => {
-      // @ts-test-if: provider === 'mongodb'
-      const result = prisma.$transaction(
-        async (tx) => {
-          await tx.user.create({ data: { email: 'user@example.com' } })
-        },
-        {
-          // @ts-test-if: provider !== 'mongodb'
-          isolationLevel: 'NotAValidLevel',
+          expect(await didLogCommit).toEqual(true)
```

Comment:
It would be nice to check what exactly is logged, is there a specific reason why this test is more minimal than the original one?